gitpod-io/observability
# Add cardinality-exporter sidecar to Prometheus

## Description

This PR adds [cardinality-exporter](https://github.com/ArthurSens/cardinality-exporter) as a sidecar to Prometheus.

The cardinality-exporter will hit the endpoint http://localhost:9090/api/v1/status/tsdb and expose the response as Prometheus metrics. Those metrics will be used to create Grafana dashboards that will support teams in understanding what are the most expensive metrics today.

## How to test


- Deploy the stack with `make deploy-satellite-go`. (It will try to start a KinD cluster, so won't work inside Gitpod)
- Expose Prometheus UI with `kubectl port-forward -n monitoring-satellite prometheus-k8s-0 9090`
- Execute the query `{container="cardinality-exporter"}`, it should return metrics from the new exporter
